src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -2121,9 +2121,22 @@ void AwtWindow::SetIconData(JNIEnv* env, jintArray iconRaster, jint w, jint h,
         hOldIconSm = m_hIconSm;
     }
     m_hIconSm = NULL;
-    m_hIcon = CreateIconFromRaster(env, iconRaster, w, h);
-    JNU_CHECK_EXCEPTION(env);
-    m_hIconSm = CreateIconFromRaster(env, smallIconRaster, smw, smh);
+
+    try {
+        m_hIcon = CreateIconFromRaster(env, iconRaster, w, h);
+        JNU_CHECK_EXCEPTION(env); // Might throw here
+        m_hIconSm = CreateIconFromRaster(env, smallIconRaster, smw, smh);
+        JNU_CHECK_EXCEPTION(env); // Or here
+    } catch (...) {
+        // Clean up any allocated resources here
+        if (m_hIcon != NULL) {
+            DestroyIcon(m_hIcon);
+        }
+        if (m_hIconSm != NULL) {
+            DestroyIcon(m_hIconSm);
+        }
+        throw; // Re-throw the exception
```

Comment:
The C++ exception shouldn't be *re-thrown* here.

Whether we should raise a Java exception to indicate an error is to be discussed. If we restore the state of the `AwtWindow` object as if no failure occurred, it can continue. And I don't see any meaningful way to return an error back to Java here.

`CreateIconFromRaster` throws `NullPointerException` if it fails to get the image raster.

https://github.com/openjdk/jdk/blob/e413fc643c4a58e3c46d81025c3ac9fbf89db4b9/src/java.desktop/windows/native/libawt/windows/awt_Window.cpp#L2076-L2078

where `JNI_CHECK_NULL_GOTO` is

https://github.com/openjdk/jdk/blob/27646e551686ec02740600fc73694fc2fbd00a88/src/java.desktop/windows/native/libawt/windows/awt.h#L48-L54